templates/template_helpers.go

```diff
@@ -400,6 +402,20 @@ func slice(start interface{}, end interface{}, increment interface{}) ([]int, er
 	return out, nil
 }
 
+// Return the keys in the given map. This method always returns the keys in sorted order to provide a stable iteration
+// order.
+func keys(m map[string]string) []string {
```

Comment:
It's a "logic less" templating language, so you don't have access to the full power of a programming language. With such languages, you either need EVERYTHING to be meticulously filled out in your data blob, which in practical terms never works, or a huge slew of helpers, which it does not include. I've already had to add over a dozen helpers for basic tasks like arithmetic, text transforms, and even support for basic looping. If we stick with Go templating, to make boilerplate flexible and powerful, we'll have to add dozens more. 

I used to be a believer in logic less templating, especially after seeing what PHP devs would do in their "templates" (e.g. database calls), but the truth is that views DO have logic and SHOULD have logic and taking away a developer's tools for dealing with that logic is way too extreme of a solution.